import { AppError } from "app/helpers/app-error";
import * as z from "zod";

export const SchemaCreate = z.object({
  // name: z.string().min(1).max(255),
});

export type SchemaType = z.infer<typeof SchemaCreate>;

export async function create(data: SchemaType) {
  try {
    SchemaCreate.parse(data);
  } catch (error) {
    if (error instanceof z.ZodError || error instanceof AppError) {
      throw error;
    }

    throw AppError.E_GENERIC_ERROR("Unexpected error during validation.");
  }
}
